assets/js/blocks/classic-template/index.tsx

```diff
@@ -99,51 +105,128 @@ const Edit = ( { clientId, attributes }: Props ) => {
- */
-registerBlockType( 'woocommerce/legacy-template', {
-	title: __( 'WooCommerce Classic Template', 'woo-gutenberg-products-block' ),
-	icon: (
-		<Icon icon={ box } className="wc-block-editor-components-block-icon" />
-	),
-	category: 'woocommerce',
-	apiVersion: 2,
-	keywords: [ __( 'WooCommerce', 'woo-gutenberg-products-block' ) ],
-	description: __(
-		'Renders classic WooCommerce PHP templates.',
-		'woo-gutenberg-products-block'
-	),
-	supports: {
-		align: [ 'wide', 'full' ],
-		html: false,
-		multiple: false,
-		reusable: false,
-		inserter: false,
-	},
-	example: {
-		attributes: {
-			isPreview: true,
+let currentValue: string | undefined;
+const templates = Object.keys( TEMPLATES );
+
+const registerClassicTemplateBlock = ( template?: string ) => {
+	registerBlockType( BLOCK_SLUG, {
+		title: template
+			? TEMPLATES[ template ].title
+			: __(
+					'WooCommerce Classic Template',
+					'woo-gutenberg-products-block'
+			  ),
+		icon: (
+			<Icon
+				icon={ box }
+				className="wc-block-editor-components-block-icon"
+			/>
+		),
+		category: 'woocommerce',
+		apiVersion: 2,
+		keywords: [ __( 'WooCommerce', 'woo-gutenberg-products-block' ) ],
+		description: __(
+			'Renders classic WooCommerce PHP templates.',
+			'woo-gutenberg-products-block'
+		),
+		supports: {
+			align: [ 'wide', 'full' ],
+			html: false,
+			multiple: false,
+			reusable: false,
+			inserter: true,
 		},
-	},
-	attributes: {
-		/**
-		 * Template attribute is used to determine which core PHP template gets rendered.
-		 */
-		template: {
-			type: 'string',
-			default: 'any',
+		example: {
+			attributes: {
+				isPreview: true,
+			},
+		},
+		attributes: {
+			/**
+			 * Template attribute is used to determine which core PHP template gets rendered.
+			 */
+			template: {
+				type: 'string',
+				default: 'any',
+			},
+			align: {
+				type: 'string',
+				default: 'wide',
+			},
 		},
-		align: {
-			type: 'string',
-			default: 'wide',
+		edit: ( { attributes, clientId } ) => {
+			return (
+				<Edit
+					attributes={ {
+						...attributes,
+						template: template ?? ( attributes.template as string ),
+					} }
+					clientId={ clientId }
+				/>
+			);
 		},
-	},
-	edit: Edit,
-	save: () => null,
-} );
+		save: () => null,
+	} );
+};
+
+const isClassicTemplateBlockRegisteredWithAnotherTitle = (
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	block: Block< any > | undefined,
+	parsedTemplate: string | undefined
+) => parsedTemplate && block?.title !== TEMPLATES[ parsedTemplate ].title;
+
+const hasTemplateSupportForClassicTemplateBlock = (
+	parsedTemplate: string | undefined
+) => parsedTemplate && templates.includes( parsedTemplate );
+
+// @todo Refactor when there will be possible to show a block according on a template/post with a Gutenberg API. https://github.com/WordPress/gutenberg/pull/41718
+
+if ( isExperimentalBuild() ) {
+	subscribe( () => {
```

Comment:
Looks like this is being called too often. Do you think that we should debounce it?

Reply:
I don't think that we have space for improvements here :'( 

`subscribe` is triggered every time that an action is dispatched ([docs](https://redux.js.org/api/store#subscribelistener)). The check that I did [here](https://github.com/woocommerce/woocommerce-blocks/blob/6c227af2eec957a785659bd37412ba7b016523f5/assets/js/blocks/classic-template/index.tsx/#L191-L193), it useful for not doing not necessary computation. Do you have any particular ideas?


 BTW, this is should be a workaround until this [PR](https://github.com/WordPress/gutenberg/pull/41718) (or the feature) will land on Gutenberg. If you read the description of the PR that I linked, you can see that for now, [this is the only way to go.](https://github.com/WordPress/gutenberg/issues/41062)

Reply:
What do you think about limiting the Classic Template Block to the Site Editor only?

Reply:
> What do you think about limiting the Classic Template Block to the Site Editor only?

Theoretically, this check is done [here](https://github.com/woocommerce/woocommerce-blocks/pull/6539/files/6c227af2eec957a785659bd37412ba7b016523f5..1ac90a3b68cc5dd17a9b4e35260db8e4d01f1512#diff-49cf8439043f78ee9f68b6374ef821be0b59c5804fec85e9aecf528d7fe9bcecR197-R198). If your idea is always on frontend side, I'm pretty sure that we should use always the `subscribe` function because we need to read the value in the state at every update.

Reply:
@gigitux I'm sorry I wasn't very clear! I mean only controlling the block loading on the PHP side. Like what we did (and removed) with the Mini Cart block, see https://github.com/woocommerce/woocommerce-blocks/pull/6351/files.